Assets/SteamVR_Unity_Toolkit/Scripts/Controls/2D/RadialMenu/RadialMenu.cs

```diff
@@ -0,0 +1,300 @@
+        {
+            transform.localScale += Dir * Time.deltaTime * 4f; //Tweening function - currently 0.25 second linear
+            yield return true;
+            i++;
+        }
+        transform.localScale = Dir * targetScale;
+        StopCoroutine("TweenMenuScale");
+    }
+
+    #endregion
+
+    #region Generation
+    
+    //Creates all the button Arcs and populates them with desired icons
+    public void RegenerateButtons()
+    {
+        RemoveAllButtons();
+        for (int i = 0; i < Buttons.Count; i++)
+        {
+            // Initial placement/instantiation
+            GameObject newButton = (GameObject)Instantiate(buttonPrefab);
+            newButton.transform.SetParent(transform);
+            newButton.transform.localScale = Vector3.one;
+            newButton.GetComponent<RectTransform>().offsetMax = Vector2.zero;
+            newButton.GetComponent<RectTransform>().offsetMin = Vector2.zero;
+
+            //Setup button arc
+            UICircle circle = newButton.GetComponent<UICircle>();
+            if (buttonThickness == 1)
+                circle.fill = true;
+            else
+                circle.thickness = (int)(buttonThickness * ((float)GetComponent<RectTransform>().rect.width / 2f));
+            int fillPerc = (int)(100f / Buttons.Count);
+            circle.fillPercent = fillPerc;
+            circle.color = buttonColor;
+
+            //Final placement/rotation
+            float angle = ((360 / Buttons.Count) * i) + offsetRotation;
+            newButton.transform.localEulerAngles = new Vector3(0, 0, angle);
+            newButton.layer = 4; //UI Layer
+            newButton.transform.localPosition = Vector3.zero;
+            if (circle.fillPercent < 55)
+            {
+                float angleRad = (angle * Mathf.PI) / 180f;
+                Vector2 angleVector = new Vector2(-Mathf.Cos(angleRad), -Mathf.Sin(angleRad));
+                newButton.transform.localPosition += (Vector3)angleVector * offsetDistance;
+            }
+
+            //Place and populate Button Icon
+            GameObject buttonIcon = newButton.GetComponentInChildren<RadialButtonIcon>().gameObject;
+            if (Buttons[i].ButtonIcon == null)
+                buttonIcon.SetActive(false);
+            else
+            {
+                buttonIcon.GetComponent<Image>().sprite = Buttons[i].ButtonIcon;
+                buttonIcon.transform.localPosition = new Vector2(-1 * ((newButton.GetComponent<RectTransform>().rect.width / 2f) - (circle.thickness / 2f)), 0);
+                //Min icon size from thickness and arc
+                float scale1 = Mathf.Abs(circle.thickness);
+                float R = Mathf.Abs(buttonIcon.transform.localPosition.x);
+                float bAngle = (359f * circle.fillPercent * 0.01f * Mathf.PI) / 180f;
+                float scale2 = (R * 2 * Mathf.Sin(bAngle / 2f));
+                if (circle.fillPercent > 24) //Scale calc doesn't work for > 90 degrees
```

Comment:
I'm gonna stop saying braces now :) but i think it's worth having them around every conditional block just because it's done everywhere else